src/components/orderDetails/OrderDetails.tsx

```diff
@@ -0,0 +1,345 @@
+      `${URL_ORDER_DETAILS}/${orderId}`,
+      {},
+    );
+    setOrderData(data || {});
+    if (data?.commerceItem?.length) {
+      const convertedArray: CommerceItemData[] = data.commerceItem.map(
+        (item: CommerceItem) => {
+          return {
+            SKU: item.catalogRefId,
+            Type: item.itemClassType,
+            Quantity: item.quantity,
+            "Unit Price": item.priceInfo.listPrice,
+            "Total Price": item.priceInfo.rawTotalPrice,
+          };
+        },
+      );
+      setItemTableData(convertedArray);
+    }
+    setIsLoading(false);
+  };
+
+  const getOmsOrderStatus = async () => {
+    const data: OmsOrderStatus = await fetchData(
+      `${URL_OMS_ORDER_STATUS}/${orderId}`,
+      {},
+    );
+    setOmsOrderStatus(data || {});
+  };
+
+  const getOmsOrderFlow = async () => {
+    const data: OmsOrderFlow = await fetchData(
+      `${URL_OMS_ORDER_FLOW}/${orderId}`,
+      {},
+    );
+    console.log(omsOrderFlow);
+    setOmsOrderFlow(data || {});
+  };
+
+  const getPromotions = async () => {
+    const promoUrl = URL_PROMOTIONS.replace(":orderId", orderId);
+    const data: OmsOrderFlow = await fetchData(promoUrl, {});
+    console.log(promotions);
+    setPromotions(data || {});
+  };
+
+  const formatName = (fullName): string => {
+    return fullName
+      .split(" ")
+      .map((part) => part.charAt(0).toUpperCase() + part.slice(1).toLowerCase())
+      .join(" ");
+  };
+
+  const showPromotions = (): void => {
+    //show promotion modal
+  };
+
+  const showOmsStatusInfo = (): void => {
+    //show oms modal
+  };
+
+  const showOrderTimeline = (): void => {
+    //show order timeline modal
+  };
+
+  const viewAllItems = (): void => {
+    //viewAll functionality
+  };
+
+  return isLoading ? (
+    <Loader />
+  ) : Object.keys(orderData).length > 0 ? (
+    <div id="orderDetailsComp">
+      <div className="grid gap-y-0 grid-cols-1 m-4 p-0 sm:py-4 sm:px-6 sm:bg-[#30343B] rounded-md">
+        <div className="flex justify-between border-none !bg-[#1C1C20] sm:!bg-inherit">
+          <span className="w-1/2 !text-lg !text-[#F2F2F2] font-bold !bg-[#1C1C20] sm:!bg-inherit">
+            {ORDER_DETAILS}
+          </span>
+          <span
+            className="w-1/2 justify-end flex items-center !text-[12px] font-normal cursor-pointer !bg-[#1C1C20] sm:!bg-inherit"
+            onClick={showPromotions}
+          >
+            <CustomIcon
+              className="mr-1"
+              alt="promotion-icon"
+              src={PromotionsIcon}
+              width="0.75rem"
+              height="0.75rem"
+            />
+            {PROMOTIONS}
+          </span>
+        </div>
+        <div className="flex flex-col border-t-0 sm:border-t border-solid border-[#383F47] sm:grid sm:gap-y-0 sm:grid-cols-2 ">
+          <div className="flex justify-between w-full py-1 px-4 sm:p-0 border-b border-solid border-[#383F47] sm:border-none justify-start bg-[#30343B] rounded-t-md">
```

Comment:
M still few lines of long css being repeated again and again.
Ex:
`flex justify-between w-full py-1 px-4 sm:p-0 border-b border-solid border-[#383F47] sm:border-none`

Consider creating a wrapper classname for such long repetitive classes so that it is easy to maintain and code also looks cleaner.

in this case, u can do this in `index.css`
```
.flexWrapper {
   @apply flex justify-between w-full py-1 px-4 sm:p-0 border-b border-solid border-[#383F47] sm:border-none
}
```

Reply:
Done